### ISSUE TITLE
Microsoft.NETCore.Jit package has asset for .NET Native projects.

### ISSUE THREAD

**Issue by weshaggard** [dotnet/coreclr#5714]
I was testing the UWP meta-package which includes a reference to Microsoft.NETCore.Runtime.CoreCLR which also references Microsoft.NETCore.Jit so it causes the clrjit.dll to be dumped out into an application folder for .NET Native. I don't know if it hurts anything yet but it certainly isn't needed and might cause trouble in some cases in the future.

I suggest doing what we did for the coreclr package here with the jit package and add placeholders for aot. See https://github.com/dotnet/coreclr/blob/master/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/win/Microsoft.NETCore.Runtime.CoreCLR.pkgproj#L43

**Comment by weshaggard** [comment]
cc @ericstj 

@pgavlin I'm assigning to you because you created this package but feel free to re-assign.

**Comment by morganbr** [comment]
clrjit.dll is almost certainly not WACK compliant, so if an app ended up with it, it probably couldn't get through the Store.

**Comment by ericstj** [comment]
Right, which is why this needs to be fixed...

**Comment by bricelam** [comment]
Confirmed. It fails WACK.

> File bin\x64\Release\AppX\clrjit.dll has failed the AppContainerCheck check.

    API SystemFunction036 in advapi32.dll is not supported for this application type. clrjit.dll calls this API.
    API DebugBreak in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API HeapValidate in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API LoadLibraryExW in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API RtlCaptureContext in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API RtlVirtualUnwind in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API TlsAlloc in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API TlsFree in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API TlsGetValue in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API TlsSetValue in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API UnhandledExceptionFilter in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API VirtualAlloc in kernel32.dll is not supported for this application type. clrjit.dll calls this API.
    API VirtualProtect in kernel32.dll is not supported for this application type. clrjit.dll calls this API.

**Comment by bricelam** [comment]
My `project.json` is pretty simple too.
```JSON
{
  "dependencies": {
    "Microsoft.EntityFrameworkCore.Sqlite": "1.0.0",
    "Microsoft.EntityFrameworkCore.Tools": {
      "version": "1.0.0-preview2-final",
      "type": "build"
    },
    "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.0",
    "MvvmLightLibs": "5.3.0"
  },
  "frameworks": {
    "uap10.0": {}
  },
  "runtimes": {
    "win10-arm": {},
    "win10-arm-aot": {},
    "win10-x86": {},
    "win10-x86-aot": {},
    "win10-x64": {},
    "win10-x64-aot": {}
  }
}
```

**Comment by rowanmiller** [comment]
I see this is triaged for 1.1.0, but doesn't this mean that anyone who uses the 5.2.0 version of the UWP meta package won't be able to publish their app? This prevents using Entity Framework Core because we require 5.2.0.

**Comment by natemcmaster** [comment]
This is more than just an EF issue though, right? Isn't Microsoft.NetCore.UniversalWindowsPlatform v5.2.0 RTM-ing on Monday?

**Comment by weshaggard** [comment]
We are shipping a stable 5.2.0 UWP meta-package on Monday, however it currently will not be the default for any new projects and users would need to explicitly reference it only if they need to depend on a newer netstandard based package. I don't expect much usage of this meta-package.

As for this particular issue there is a workaround by adding the following to your dependencies section in your project.json:
```
    "runtime.win7-x64.Microsoft.NETCore.Jit": {
      "version": "1.0.2",
      "exclude": "native"
    },
    "runtime.win7-x86.Microsoft.NETCore.Jit": {
      "version": "1.0.2",
      "exclude": "native"
    }
```

**Comment by pgavlin** [comment]
The necessary changes to the JIT packages have been checked in. Can we close this issue, or is it tracking additional work?

**Comment by weshaggard** [comment]
Fixed by #5959
